src/main/java/org/ossgang/commons/observables/operators/CombineLatestOperators.java

```diff
@@ -0,0 +1,169 @@
+        return combineLatestObjects(toIndexMap(sources), idxMap -> combiner.apply(fromIndexMap(idxMap)));
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatestObjects(Map)
+     */
+    public static <K> ObservableValue<Map<K, Object>> combineLatestObjects(Map<K, ? extends Observable<?>> sourcesMap) {
+        return combineLatestObjects(sourcesMap, Function.identity());
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatestObjects(List)
+     */
+    public static ObservableValue<List<Object>> combineLatestObjects(List<? extends Observable<?>> sources) {
+        return combineLatestObjects(sources, Function.identity());
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatest(Map, Function)
+     */
+    public static <K, I, O> ObservableValue<O> combineLatest(Map<K, ? extends Observable<I>> sourcesMap,
+                                                             Function<Map<K, I>, O> combiner) {
+        Function<Map<K, Object>, Map<K, I>> typedTranslator = values -> applyToMapValues(values, v -> (I) v);
+        return combineLatestObjects(sourcesMap, typedTranslator.andThen(combiner));
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatest(List, Function)
+     */
+    public static <I, O> ObservableValue<O> combineLatest(List<? extends Observable<I>> sources,
+                                                          Function<List<I>, O> combiner) {
+        Function<List<Object>, List<I>> typedTranslator = values -> values.stream().map(v -> (I) v).collect(toList());
+        return combineLatestObjects(toIndexMap(sources), idxMap -> typedTranslator.andThen(combiner).apply(fromIndexMap(idxMap)));
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatest(Map)
+     */
+    public static <K, I> ObservableValue<Map<K, I>> combineLatest(Map<K, ? extends Observable<I>> sourcesMap) {
+        Function<Map<K, Object>, Map<K, I>> typedTranslator = values -> applyToMapValues(values, v -> (I) v);
+        return combineLatestObjects(sourcesMap, typedTranslator);
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatest(List)
+     */
+    public static <I> ObservableValue<List<I>> combineLatest(List<? extends Observable<I>> sources) {
+        Function<List<Object>, List<I>> typedTranslator = values -> values.stream().map(v -> (I) v).collect(toList());
+        return combineLatestObjects(sources, typedTranslator);
+    }
+
+    /**
+     * @see org.ossgang.commons.observables.Observables#combineLatest(Observable, Observable, BiFunction)
+     */
+    public static <I1, I2, O> ObservableValue<O> combineLatest(Observable<I1> source1, Observable<I2> source2,
+                                                               BiFunction<I1, I2, O> combiner) {
+        List<Observable<Object>> sourcesListObject = Arrays.asList(
```

Comment:
Could we do `List<Observable<?>>` to save the (quite verbose) up-casting from `Observable<Ix>` to `Observable<Object>`?

Reply:
Absolutely ! Leftover from API reshuffling 😇